src/threshold-ts/tbtc/index.ts

```diff
@@ -412,4 +429,26 @@ export class TBTC implements ITBTC {
       eventName: "OptimisticMintingCancelled",
     })
   }
+
+  get bridgeContract() {
+    return this._bridgeContract
+  }
+
+  get vaultContract() {
+    return this._tbtcVault
+  }
+
+  get tokenContract() {
+    return this._token
+  }
+
+  buildDepositKey = (
+    fundingTxHash: string,
+    fundingOutputIndex: number
+  ): string => {
+    return EthereumBridge.buildDepositKey(
+      TransactionHash.from(fundingTxHash).reverse(),
+      fundingOutputIndex
+    )
+  }
```

Comment:
I think that we should not reverse the txHash inside the `buildDepositKey` methods. I believe that we are reversing it because the transaction hash returned from `DepositRevealed` event is already reversed so we have to "unreverse" it, because it's gonna be reversed inside `EthereumBridge.buildDepositKey` function. This is specific situation that is related more to a `_findAllRevealedDeposits` function than `buildDepositKey` function. In the future we could use the `buildDepositKey` many times and the tx hash is not gonna be reversed. I think we could pass a `TransactionHash` instance here:

```suggestion
  buildDepositKey = (
    fundingTxHash: TransactionHash,
    fundingOutputIndex: number
  ): string => {
    return EthereumBridge.buildDepositKey(fundingTxHash, fundingOutputIndex)
  }
```

and then reverse it in `_findAllRevealedDeposits`:
```
  const depositKey = this.buildDepositKey(
          TransactionHash.from(fundingTxHash).reverse(),
          fundingOutputIndex
        )
```

Let me know what you think.

Reply:
The main idea was to not repeat `TransactionHash.from(fundingTxHash).reverse()`. In the Ethereum world, we can only get the `fundingTxHash` in [native Bitcoin little-endian format](https://github.com/keep-network/tbtc-v2/blob/main/solidity/contracts/bridge/Deposit.sol#L227-L228) from an event so that's why I wrapped the `EthereumBridge.buildDepositKey` into our implementation. But on the other hand, maybe someone wants to build a deposit key from the bitcoin tx hash and this wrapper won't work. So I'm not sure which direction we should go 🤔.

Reply:
19ba5b4